sublime_jedi/go_to.py

```diff
@@ -43,11 +44,55 @@ def _jump_to_in_window(self, filename, line_number=None, column_number=None, tra
                 self.view.show(self.point)
                 return
             filename, line_number, column_number = self.options[filename]
+
+        flags = self.prepare_layout(active_window, transient, filename)
+        active_window.open_file('%s:%s:%s' % (filename, line_number or 0,
+                                column_number or 0), flags)
+
+    def prepare_layout(self, window, transient, filename):
+        """
+        prepares the layout of the window to configured and returns a flags
+        for opening the file
+        """
         flags = sublime.ENCODED_POSITION
         if transient:
             flags |= sublime.TRANSIENT
-        active_window.open_file('%s:%s:%s' % (filename, line_number or 0,
-                                column_number or 0), flags)
+        goto_layout = get_settings_param(self.view, 'sublime_goto_layout')
+        if goto_layout == 'single-transient' and not transient:
+            flags |= sublime.TRANSIENT
+        elif goto_layout == 'two-panel':
+            self.switch_to_two_panel_layout(window, filename)
+        elif goto_layout == 'two-panel-transient':
+            self.switch_to_two_panel_layout(window, filename)
+            if not transient:
+                flags |= sublime.TRANSIENT
+        return flags
+
+    def switch_to_two_panel_layout(self, window, filename):
+        curr_group = window.active_group()
+        layout = window.get_layout()
+        if len(layout['cells']) == 1:
+            # currently a single panel layout so switch to two panels
+            window.set_layout({
+                'cols': [0.0, 0.5, 1.0],
+                'rows': [0.0, 1.0],
+                'cells': [[0, 0, 1, 1], [1, 0, 2, 1]],
+            })
+            # select non current group(panel)
+        selected_group = None
+        for group in range(window.num_groups()):
+            if group != curr_group:
+                selected_group = group
+                window.focus_group(group)
+                break
+        # if the file is already opened and is in current group
+        # move it to another panel. or we shouldn't ?
+        files_in_curr_group = dict([
+            (i.file_name(), i) for i in
+            window.views_in_group(curr_group)
+        ])
+        if filename and filename in files_in_curr_group:
```

Comment:
that check is there so that we wont move temporal buffer in case its transient or new file AND for some reason backend returned empty filename